<?xml version="1.0" encoding="UTF-8"?>

<!-- This launch file is used to launch a node of type ocam_ros developed by James Jackson. The project repo is located at https://github.com/superjax/ocam_ros -->


<launch>


  <arg name="device_path"          default="/dev/video0"/>   <!-- Serial Port to connect to -->
  <arg name="frame_id"             default="camera"/>        <!-- Frame id of camera measurments -->
  <arg name="camera_info_url"      default="file://$(find ocam_ros)/param/ocam.yaml"/> 
  <arg name="width"                default="640"/>           <!-- Image width in pixel -->
  <arg name="height"               default="480"/>           <!-- Image height in pixel -->
  <arg name="fps"                  default="30"/>            <!-- Camera frame rate -->
  <arg name="color"                default="true"/>         <!-- Publish the image in color if true else gray --> 
  <arg name="image_topic"          default="image"/>         <!-- The image topic name -->
  <arg name="show_image"           default="true"/>         <!-- Shows the image and enables keyboard shortcuts to adjust image brightness and exposure -->
  <arg name="rescale_camera_info"  default="false"/>     <!-- Whether or not to rescale the camera intrinsics -->
  <arg name="brightness"           default="80"/>            <!-- Adjust image brightness between values 1-127 -->
  <arg name="exposure"             default="60"/>            <!-- Adjusts camera shutter speed between value 1-625 -->
  <arg name="auto_exposure"        default="true"/>          <!-- Whether or not to automatically adjust exposure -->


  <node pkg="ocam_ros" type="ocam_ros_node" name="ocam_ros_node" output="screen">
    <param name="device_path"         type="str"     value="$(arg device_path)"/>   
    <param name="frame_id"            type="str"     value="$(arg frame_id)"/>       
    <param name="camera_info_url"     type="str"     value="$(arg camera_info_url)"/> 
    <param name="width"               type="int"     value="$(arg width)"/>          
    <param name="height"              type="int"     value="$(arg height)"/>           
    <param name="fps"                 type="int"     value="$(arg fps)"/>            
    <param name="color"               type="bool"    value="$(arg color)"/>         
    <param name="image_topic"         type="str"     value="$(arg image_topic)"/>         
    <param name="show_image"          type="bool"    value="$(arg show_image)"/>         
    <param name="rescale_camera_info" type="bool"    value="$(arg rescale_camera_info)"/>     
    <param name="brightness"          type="int"     value="$(arg brightness)"/>            
    <param name="exposure"            type="int"     value="$(arg exposure)"/>            
    <param name="auto_exposure"       type="bool"    value="$(arg auto_exposure)"/>    
  </node>      




</launch>
